src/main/java/com/canelmas/kafka/connect/FieldAndTimeBasedPartitioner.java

```diff
@@ -90,17 +90,27 @@ public String extract(final ConnectRecord<?> record) {
             final StringBuilder builder = new StringBuilder();
 
             for (final String fieldName : this.fieldNames) {
+                String field;
```

Comment:
String field;
String partitionName;

String[] fields = fieldName.split(":");
if(fields.length < 1) {
    field = fields[0];
    partitionName = fields[1];
 } else  {
    field = partitionName = fields[0];
}
I think it's more efficient using length that contains, but only it's an opinion, feel free to leave it as you want.

Reply:
Agree with you about efficiency